app/assets/javascripts/galleries/add_new.js

```diff
@@ -86,8 +86,9 @@ function bindFileInput(fileInput) {
         submitButton.prop('disabled', false);
 
         // extract key and generate URL from response
+        // replace https://glowfic-constellation.s3.amazonaws.com with Cloudfront URL
         var key   = $(data.jqXHR.responseXML).find("Key").text();
-        var url   = 'https://d1anwqy6ci9o1i.cloudfront.net/' + key;
+        var url   = $(data.jqXHR.responseXML).find("Location").text().replace(form.data('url'), 'https://d1anwqy6ci9o1i.cloudfront.net');
```

Comment:
I also got this (if you try the link I gave) - it seems to *work* just fine and so I think that might be what it's supposed to be, just it works if you don't escape those characters.